[IMP] blocked_location Adds making the blocked reason required (story…

BlommaertsEdwin opened the pull request:
…/14430)

- Makes the blocked reason required on the interface UI of Odoo
- Executes a migration script setting a default reasons for currently blocked locations with no reason given.

PeterAlabaster commented on addons/blocked_locations/migrations/0.2/pre-migrate.py:
```diff
@@ -0,0 +1,3 @@
+def migrate(cr, version):
+    # Update blocked reasons with dummy reasons since we're making this field a required field if the location is blocked.
+    cr.execute("UPDATE stock_location SET u_blocked_reason = 'Location is blocked' WHERE u_blocked IS True and u_blocked_reason IS Null;")
```
nit: format this more like:
```python
    cr.execute("""
        UPDATE stock_location
        SET u_blocked_reason = 'Location is blocked'
        WHERE u_blocked IS True and u_blocked_reason IS Null;
    """) 
```

RHODOK replied:
and a new line at the end =) to get rid of the angry red minus sign =D

williamsajr commented on addons/blocked_locations/migrations/0.2/pre-migrate.py:
```diff
@@ -0,0 +1,7 @@
+def migrate(cr, version):
+    # Update blocked reasons with dummy reasons since we're making this field a required field if the location is blocked.
```
Nit: use """ for docstrings

williamsajr replied:
Nit for fix: Line length is now 128 characters

williamsajr commented on addons/blocked_locations/models/stock_location.py:
```diff
@@ -3,67 +3,74 @@
                     raise ValidationError(
-                            _('Location cannot be blocked because '
-                              'it contains reserved stock.'))
+                        _("Location cannot be blocked because " "it contains reserved stock.")
```
Nit: Don't need two strings here (black formatting doesn't combine strings when removing line breaks)

williamsajr commented on addons/blocked_locations/tests/test_location.py:
```diff
@@ -3,99 +3,116 @@
+        self.assertEqual(
+            e.exception.name,
+            "A reason for blocking the locations is required when attempting to block a location.",
+        )
```
It's probably worth adding a test to explicitly check that when we don't supply a message that an exception is raised

BlommaertsEdwin replied:
Hi @williamsajr isn't that what this test already does? :)

williamsajr replied:
Yep you're right, I get confused before my first cup of coffee in the morning

williamsajr commented on addons/blocked_locations/models/stock_location.py:
```diff
@@ -3,67 +3,74 @@
-                    _('%s %s Please speak to a team leader '
-                      'to resolve the issue.') %
-                      (prefix, ' '.join(blocked_locations._prepare_blocked_msg())))
+                _("%s %s Please speak to a team leader " "to resolve the issue.")
```
Another double string here

BlommaertsEdwin committed:
[IMP] blocked_location Adds making the blocked reason required (story/14430)

- Makes the blocked reason required on the interface UI of Odoo
- Adds contraint that a blocked reason needs to be specified
- Executes a migration script setting a default reasons for currently blocked locations with no reason given.

apatrick88 commented:
Bit late now for this one, but can we keep formatting changes in a separate commit please? It makes code reviews much easier! Thanks
